externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/EventManager.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.spark.events
+
+import org.apache.spark.kyuubi.SparkContextHelper
+import org.apache.spark.scheduler.SparkListenerEvent
+
+object EventManager {
+
+  private def postListenerEvent(event: SparkListenerEvent): Unit = {
+    SparkContextHelper.postListenerEvent(event)
+  }
+
+  def onSessionCreated(ip: String, sessionId: String, userName: String = "UNKNOWN"): Unit = {
+    postListenerEvent(KyuubiEngineSessionCreatedEvent(ip, sessionId,
+      userName, System.currentTimeMillis()))
+  }
+
+  def onSessionClosed(sessionId: String): Unit = {
+    postListenerEvent(KyuubiEngineSessionClosedEvent(sessionId,
+      System.currentTimeMillis()))
+  }
+
+  def onStatementStart(
+    id: String,
+    sessionId: String,
+    statement: String,
+    groupId: String,
+    userName: String = "UNKNOWN"): Unit = {
+    postListenerEvent(KyuubiEngineOperationStartEvent(id, sessionId, statement, groupId,
+      System.currentTimeMillis(), userName))
+  }
+
+  def onStatementParsed(id: String, executionPlan: String): Unit = {
+    postListenerEvent(KyuubiEngineOperationParsedEvent(id, executionPlan))
+  }
+
+  def onStatementCanceled(id: String): Unit = {
+    postListenerEvent(KyuubiEngineOperationCanceledEvent(id, System.currentTimeMillis()))
+  }
+
+  def onStatementTimeout(id: String): Unit = {
+    postListenerEvent(KyuubiEngineOperationTimeoutEvent(id, System.currentTimeMillis()))
+  }
+
+  def onStatementError(id: String, errorMsg: String, errorTrace: String): Unit = {
+    postListenerEvent(KyuubiEngineOperationErrorEvent(id, errorMsg, errorTrace,
+      System.currentTimeMillis()))
+  }
+
+  def onStatementFinish(id: String): Unit = {
+    postListenerEvent(KyuubiEngineOperationFinishEvent(id, System.currentTimeMillis()))
+  }
+
+  def onOperationClosed(id: String): Unit = {
+    postListenerEvent(KyuubiEngineOperationClosedEvent(id, System.currentTimeMillis()))
+  }
+}
+
+case class KyuubiEngineSessionCreatedEvent(
```

Comment:
It's duplicate with `org.apache.kyuubi.engine.spark.events.SparkStatementEvent`?

Reply:
yes，we can use the current events directly

Reply:
> we can use the current events directly

`ExecutionInfo` has more fields than `SparkStatementEvent`, can not use current events directly.

We can make `EventLoggingService` as a sparklistener, then reuse `SessionInfo` and `ExecutionInfo`.